web/src/main/java/com/microsoft/applicationinsights/web/internal/correlation/AppProfileFetcher.java

```diff
@@ -21,14 +21,16 @@
 
 package com.microsoft.applicationinsights.web.internal.correlation;
 
+import java.io.Closeable;
 import java.io.IOException;
 import java.util.concurrent.ExecutionException;
+
 import org.apache.http.ParseException;
 
 /**
  * Retrieves the application profile from storage
  */
- public interface AppProfileFetcher {
+ public interface AppProfileFetcher extends Closeable {
```

Comment:
For this @littleaj could you please confirm if this doesn't break spring boot scenario. Specifically I would like to see that with this change do we still need to register the Bean for WebRequestTrackingFilter or the framework can automatically pick it?

Reply:
Sounds good. I'll see what I can find out.